contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -74,15 +71,37 @@ contract FraudProofFacet is StateChannelCommon {
         return signer1; 
     }
 
-    function _handleBlockEmptyBlock(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) pure internal returns (address) {
+    function _handleBlockEmptyBlock(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
         BlockEmptyProof memory blockEmptyProof = abi.decode(encodedProof, (BlockEmptyProof));
+        StateSnapshot memory latestStateSnapshot = blockEmptyProof.latestStateSnapshot;
+        bytes32 previousStateSnapshotHash = blockEmptyProof.previousStateSnapshotHash;
+        bytes memory previousStateMachineState = blockEmptyProof.previousStateMachineState;
         Block memory fraudBlock = abi.decode(blockEmptyProof.emptyBlock.encodedBlock, (Block));
 
         if(fraudProofVerificationContext.channelId != fraudBlock.transaction.header.channelId) {
-            return address(0);
+            revert ErrorNotSameChannelId();
         }
-        if(fraudBlock.transaction.header.transactionCnt != uint(0)) {
-            return address(0);
+
+        if(fraudBlock.transaction.header.transactionCnt == 0){
+            if(previousStateSnapshotHash != keccak256(abi.encode(latestStateSnapshot))){
+                revert ErrorInvalidBlock();
+            }
+        }else{
+            // transition the state from previousStateSnapshot to latestStateSnapshot
+            (bool isTransitionValid, bytes memory blockTransitionEncodedModifiedState) = AStateChannelManagerProxy(address(this)).executeStateTransitionOnState(
+                fraudProofVerificationContext.channelId,
+                previousStateMachineState,
+                fraudBlock.transaction
+            );
+            if(!isTransitionValid){
+                revert ErrorInvalidBlockStateTransition();
+            }
+            if(latestStateSnapshot.stateMachineStateHash != keccak256(blockTransitionEncodedModifiedState)){
+                revert ErrorInvalidBlockState();
+            }
+            if(previousStateSnapshotHash != keccak256(abi.encode(latestStateSnapshot))){
+                revert ErrorInvalidStateSnapshot();
+            }
```

Comment:
We'll discuss this and everything else in the call tomorrow

Reply:
this the peer will provide the state on which we will use to transition, the prevStateSnapshotHash, the latestSnapShot, so when transitioning, the encodedModified state will be checked against latestSnapShot latestStatehash.

Where is the logic incosistency?